### PR TITLE
Port over some SP v2 bits

### PR DESCRIPTION
**Pull request by jefferai**
Specifically:

* Add too-large handling to Physical (Consul only for now)
* Contextify some identity funcs
* Update SP protos

### CODE REVIEW

**Comment by vishalnayak** [comment]
I have added size limiting to inmem storage. Please take a look. If it looks good, this is good to go.

File: physical/inmem/inmem.go [hunk 1]
```diff
@@ -42,15 +44,26 @@ type InmemBackend struct {
...
 type TransactionalInmemBackend struct {
 	InmemBackend
 }
 
 // NewInmem constructs a new in-memory backend
-func NewInmem(_ map[string]string, logger log.Logger) (physical.Backend, error) {
-	in := &InmemBackend{
+func NewInmem(conf map[string]string, logger log.Logger) (physical.Backend, error) {
+	maxValSize := 0
+	maxValSizeStr, ok := conf["max_val_size"]
```
Comment (jefferai) [inline physical/inmem/inmem.go]: nit: can this be "max_value_size"?

File: physical/inmem/inmem.go [hunk 2]
```diff
@@ -59,14 +72,24 @@ func NewInmem(_ map[string]string, logger log.Logger) (physical.Backend, error)
...
+	}, nil
 }
 
 // Basically for now just creates a permit pool of size 1 so only one operation
 // can run at a time
-func NewTransactionalInmem(_ map[string]string, logger log.Logger) (physical.Backend, error) {
-	in := &TransactionalInmemBackend{
+func NewTransactionalInmem(conf map[string]string, logger log.Logger) (physical.Backend, error) {
+	maxValSize := 0
+	maxValSizeStr, ok := conf["max_val_size"]
```
Comment (jefferai) [inline physical/inmem/inmem.go]: same here

File: physical/inmem/inmem.go (hunk 1 above)
Comment (vishalnayak) [reply physical/inmem/inmem.go]: Fixed.

File: physical/inmem/inmem.go (hunk 2 above)
Comment (vishalnayak) [reply physical/inmem/inmem.go]: Fixed.

**Pull request merged**